src/rp/util/string.clj

```diff
@@ -28,3 +28,8 @@
   (some-> s
           parse-double*
           double))
+
+(defn non-blank-string?
+  "Returns true if x is a non-blank string"
+  [x] (and (string? x)
+           (not (str/blank? x))))
```

Comment:
In general the only things that require concrete boolean values are Java API's, so no, I don't think there's anything lost by providing only fluent functions using truthy semantics and not bother with predicates. The general rule is to use the `boolean` function in place when you need to give a Java method a literal `true` or `false`.